configsuite/schema.py

```diff
@@ -86,54 +95,107 @@ def _check_required_not_default(schema_level):
     ),
     MK.Content: {
         MK.Type: {MK.Type: types.Type},
-        MK.Required: {MK.Type: types.Bool, MK.Required: False},
-        MK.AllowNone: {MK.Type: types.Bool, MK.Required: False},
-        MK.Description: {MK.Type: types.String, MK.Required: False},
-        MK.Default: {MK.Type: _Anytype, MK.Required: False},
+        MK.Required: {
+            MK.Type: types.Bool,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.Required],
+        },
+        MK.AllowNone: {
+            MK.Type: types.Bool,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.AllowNone],
+        },
+        MK.Description: {
+            MK.Type: types.String,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.Description],
+        },
+        MK.Default: {MK.Type: _Anytype, MK.Required: False, MK.AllowNone: True},
         MK.ElementValidators: {
             MK.Type: types.List,
-            MK.Required: False,
             MK.Content: {MK.Item: {MK.Type: types.Callable}},
         },
         MK.ContextValidators: {
             MK.Type: types.List,
-            MK.Required: False,
             MK.Content: {MK.Item: {MK.Type: types.Callable}},
         },
-        MK.Transformation: {MK.Type: types.Callable, MK.Required: False},
-        MK.ContextTransformation: {MK.Type: types.Callable, MK.Required: False},
-        MK.LayerTransformation: {MK.Type: types.Callable, MK.Required: False},
+        MK.Transformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
+        MK.ContextTransformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
+        MK.LayerTransformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
     },
 }
 
 
-def assert_valid_schema(schema, allow_default=False):
-    assert_valid_schema_level(schema, allow_default=allow_default)
+def assert_valid_schema(schema):
+    _assert_valid_schema(
+        _META_SCHEMA, allow_default=False, validate_named_keys=False
```

Comment:
To avoid the normal requirement of `named_dict` only having strings as keys (the meta schema has the values in `MetaKeys` as keys) the option `validate_named_keys` was introduced. This option is set to false when validating the meta schema, but set to true (which gives the original behaviour) when validating the schema provided by the user.